helm/vitalam-api/templates/configmap.yaml

```diff
@@ -0,0 +1,32 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "vitalam-api.fullname" . }}-config
+  labels:
+    {{- include "vitalam-api.labels" . | nindent 4 }}
+data:
+  port: {{ .Values.config.port | quote }}
+  logLevel: {{ .Values.config.logLevel }}
+  {{- if .Values.config.externalNodeHost }}
+  nodeHost: {{ .Values.config.externalNodeHost }}
+  {{- else if .Values.vitalamNode.enabled }}
+  nodeHost: {{ template "vitalam-node.fullname" .Subcharts.vitalamNode }}
+  {{- else if .Values.vitalamIpfs.vitalamNode.enabled }}
+  nodeHost: {{ template "vitalam-ipfs-node.fullname" .Subcharts.vitalamIpfs }}
+  {{- else }}
+  {{- fail "Must supply either externalNodeHost or enable vitalamNode or vitalamIpfs" }}
+  {{- end }}
```

Comment:
This should probably be in a template to contain the mess. It's quite hard to read the logic here. We always want a `nodeHost` so you can likely just template the rhs

Reply:
Ok not a problem.  Will do this on Monday and replicate the logic into `vitalam-ipfs` as well.